### PR TITLE
Fixes for MSVC compile and issues while running

### PR DESCRIPTION
**Pull request by caseymcc**
Changes to cmake to compile on windows, then some error checking functionality as the chucks are empty on start.

### CODE REVIEW

**Comment by nsg** [comment]
Thank you for the PR, I like the CMake changes.

I have successfully build the project under windows with CMake and VS a while ago, then of course we do not test every master commit against windows so it's possible that master is broken atm.

From what I can see you have added plenty of checks in `compute_chunk` for not yet loaded chunks, I assume you had a crash? When did that happen, directly or when you tried to move? (there is a known bug that crashes the client if you tries to jump inside a unloaded chunk).

I prefer to find and fix the reason behind the crash instead of just adding a lot of checks and return dummy data (`return shared_ptr<ChunkModelResult>();` and `return{0, 0};`).

I can help you track the problem down if you like.

File: lib/include/chunk.h [hunk 1]
```diff
@@ -12,7 +12,8 @@
 #define BLOCK_BUFFER_SIZE (CHUNK_SIZE*CHUNK_SIZE*CHUNK_SIZE*BLOCK_SIZE)
 #define BLOCKS_HEADER_SIZE 2
 
-namespace konstructs {
+namespace konstructs 
+{
```
Comment (nsg) [inline lib/include/chunk.h]: I assume this was some automatic refactoring done by your editor/IDE? Please undo these changes because they increases the size of the diff, and they also changes the coding style. Try to look at the surrounding functions coding style and use the same. This will make both the diff and the finalized code easier to read.

File: lib/include/chunk.h [hunk 2]
```diff
@@ -40,6 +42,8 @@ namespace konstructs {
         const Vector3i position;
         BlockData *blocks;
     };
+
+    typedef std::shared_ptr<ChunkData> SharedChunkData;
```
Comment (nsg) [inline lib/include/chunk.h]: Never used.

File: src/main.cpp [hunk 3]
```diff
@@ -509,8 +511,13 @@ class Konstructs: public nanogui::Screen {
...
         if(sscanf(str.c_str(), ",%u,%u,%u,%u",
-                  &column, &size, &type, &health) != 4)
-            throw std::runtime_error(str);
+            &column, &size, &type, &health)!=4)
+        {
+            std::string what="handle_belt expected ,%u,%u,%u,%u got "+str;
+
+            cout<<what;
+//            throw std::runtime_error(what);
+        }
```
Comment (nsg) [inline src/main.cpp]: Why a cout instead of a runtime_error?

File: lib/include/chunk.h (hunk 1 above)
Comment (wgv-cmccandless) [inline lib/include/chunk.h]: Yeah np, will put them back and send a new pull request. IDE gets trigger happy.

Comment (caseymcc) [inline lib/include/chunk.h]: Yeah np, will put them back and send a new pull request. IDE gets trigger happy.

File: lib/include/chunk.h (hunk 2 above)
Comment (caseymcc) [inline lib/include/chunk.h]: Habit, but with auto now days not needed as much.

File: src/main.cpp (hunk 3 above)
Comment (caseymcc) [inline src/main.cpp]: The server was spitting out only 3 values for this and considering I have no clue the issue I remarked it for the moment as in order to catch other crashes I had all exceptions turned on (didn't mean to keep it in the check-in)

**Comment by caseymcc** [comment]
As soon as the client connected to the server it began crashing. From the short time I played with it, it seems it was trying to render before the server had passed down the chunks to the client. Basically everything was empty and the code was assuming something would already be there.

After looking through the chunking code more I don't think it would be worth hunting down the issues. I am going to try on the side to build a new chunking library, when I complete it I will look into inserting it if you all want. I will create a pull request for just the cmake changes and a separate commit on my tree for the checks in the chunk (as I can't run with out them).

File: src/main.cpp (hunk 3 above)
Comment (petterarvidsson) [inline src/main.cpp]: @caseymcc Server master/HEAD and client master/HEAD is not always in perfect sync (sorry about that, maybe we should keep them so). We are making release tags that are in sync like the latest release 6 https://github.com/konstructs/client/releases/tag/6 and https://github.com/konstructs/server/releases/tag/v0.1.17. Between them there are changes to the server <-> client protocol that are not always backwards compatible. This is one such change.

**Comment by petterarvidsson** [comment]
:+1: on the cmake changes in its own PR. Feel free to also include the use of vector instead of malloc in compute_chunk. (the function was inherited from the C-client).

For the crashes, I think the functions in world.ccp could be updated to return solid chunks when a chunk is missing, like what is done here: https://github.com/konstructs/client/blob/master/lib/src/chunk_factory.cpp#L107

Will take a look at it tomorrow evening. Thanks for reporting the bug and providing the PR!

**Comment by nsg** [comment]
I will close this now considering that the cmake changes was merged in #201 (https://github.com/konstructs/client/commit/7d538a2b39023c6afa7248aa81ec5593c7abc983) and the crash was (I hope) fixed with #200.
